src/state.cpp

```diff
@@ -580,6 +580,21 @@ State::do_export(const std::string& label,
   return _suite.expand_with_label(secret, "exporter", context, size);
 }
 
+std::vector<Credential>
+State::get_leaf_credentials() const
+{
+  std::vector<Credential> creds;
+
+  for (uint32_t i = 0; i < _tree.size().val; i++) {
+    const auto& ln = _tree.node_at(LeafIndex{ i });
```

Comment:
This is what `TreeKEMPublicKey::key_package(LeafIndex index)` is for.  `node_at` should probably be private.